### PR TITLE
fix: sort wildcard and :params 

### PR DESCRIPTION
**Pull request by zhigang1992**
We want to server user profile like https://github.com/zhigang1992 or https://twitter.com/zhigang1992 along side with screens in the app.

```typescript
const configs = {
  screens: {
    debugPanel: 'debugPanel',
    user: ':userSlug',
  },
}
getStateFromPath('/debugPanel', configs)
```

and

```typescript
const configs = {
  screens: {
    debugPanel: 'debugPanel',
    user: '*',
  },
}
getStateFromPath('/debugPanel', configs)
```

have different behaviors under Hermes engine, because we are only sorting '*'.

IMHO `:urlCode` is also a form of wildCard.

### CODE REVIEW

File: packages/core/src/__tests__/getStateFromPath.test.tsx [hunk 1]
```diff
@@ -2196,6 +2196,7 @@ it('tries to match wildcard patterns at the end', () => {
             path: '/bar/:id/',
             screens: {
               404: '*',
+              matches: ':params',
```
Comment (satya164) [inline packages/core/src/__tests__/getStateFromPath.test.tsx:2199]: Looks like a leftover?

**Review by satya164 — CHANGES_REQUESTED** [review]
Thanks for the PR. Can you add some comments similar to those that were there previously?

File: packages/core/src/__tests__/getStateFromPath.test.tsx (hunk 1 above)
Comment (zhigang1992) [reply packages/core/src/__tests__/getStateFromPath.test.tsx:2199]: No, it's meant to be here. 

that `:params` should take lower priority compare to 'test'. Just like '*'?

Comment (zhigang1992) [reply packages/core/src/__tests__/getStateFromPath.test.tsx:2199]: Since `:params` and `*` are both wildcards

Comment (satya164) [reply packages/core/src/__tests__/getStateFromPath.test.tsx:2199]: Can you rename `matches` to some name like `Test1` etc.? `matches` looks like it's some option and not a screen name

**Comment by zhigang1992** [comment]
@satya164 please take another look

**Comment by satya164** [comment]
Thank you!

**Pull request merged**